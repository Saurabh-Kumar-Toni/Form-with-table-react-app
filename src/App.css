

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.App{
  display:flex; 
  flex-direction:row;
  padding: 5px 40px 20px 50px;
}

.Form{
  padding: 20px;
  border-right: black solid 1px;
  width: 400px;
  margin-left: 50px;
  margin-top: 100px;
  text-align: center;
}
.Table{
  padding: 20px;
  border-radius: 4px;
  width: 400px;
  margin-left: 10px;
  margin-top: 100px;
  text-align: center;
}

#Table {
  width: 100%;
}

#Table table {
  width: 100%;
}

#Table td {
    border: 1px solid cadetblue;
    width: 50%;
    border-radius: 2px;
}

#Table th {
    width: 25%;
    background-color:#93CED1;
    color: white;
    border-radius: 2px;
    
}

#button {
  background-color: #93CED1;
}

#number{
  border-radius: 2px;
  width: 75%;
  margin-bottom: 10px;
  outline: none;
}

#username{
  border-radius: 2px;
  width: 75%;
  margin-bottom: 10px;
  outline: none;
}

#gender{
  border-radius: 2px;
  width: 75%;
  height: 50%;
  margin-bottom: 10px;
  outline: none;
}

#btndel{
  background: transparent;
  color: red;
}

#counter{
  text-align: center;
  width: 100%;
  
}

#heading{
  padding: 0 0 5px 20px;
  float: left;
  margin-left: 25px;
}

#searchbar{
  float: left;
  outline: none;
}