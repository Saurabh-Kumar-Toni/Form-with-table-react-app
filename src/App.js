import React, { useState } from 'react';
import './App.css';
import Form from './Component/Form/Form';
import Table from './Component/Table/table';
import Counter from './Component/Counter/counter'

const App =  () => {
  const [item, setItem] = useState([]);
  const usersList = (data) => {
    setItem([...item,data]);
  }

  const deleteUser = (data) => {
    
      let newUser = item.filter((item, id=0) => id !== data);
      setItem(newUser);
  }

  const findUser = (text) => {
    const searchedUSer = item.filter((it) => it.username.toLowerCase().includes(text.toLowerCase()));
    if(text != null)
        setItem(searchedUSer);
  }

    return (
      <div >
        
        <Counter count = {item.length}/>
        <div className="App">
        <div className="Form">
          <Form getUsers={(data) => usersList(data)} />
        </div>
        <div className="Table">
          <Table items={item} deleteUsers ={(data) => deleteUser(data)} findUsers={(text) => findUser(text)}/>
        </div>
        </div>
      </div>
    );
  
}

export default App;
