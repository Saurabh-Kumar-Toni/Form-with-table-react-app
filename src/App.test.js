import { render, screen } from '@testing-library/react';
import App from './App';

test('renders teh Application properly', () => {
  const {container} = render(<App><h2>Hello World</h2></App>);
  expect(container.firstChild).toBeInTheDocument();
});

test('renders the App Component Snapshot', () => {
 
  const {container} = render(<App />)
  expect(container.firstChild).toMatchSnapshot();
});
