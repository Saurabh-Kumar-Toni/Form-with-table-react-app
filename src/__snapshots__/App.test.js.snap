// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the App Component Snapshot 1`] = `
<div
  class="App"
>
  <div
    id="counter"
  >
    <p>
      Our Total User Count Is: 
      0
    </p>
    <p>
       Add more to increase the number
    </p>
  </div>
  <div
    class="Form"
  >
    <div
      id="Form"
    >
      <form>
        <h2>
          Hello There!!
        </h2>
        <input
          id="username"
          name="username"
          placeholder="Enter your name"
          type="text"
          value=""
        />
        <br />
        <input
          id="number"
          name="number"
          placeholder="Enter your phone number"
          type="number"
          value=""
        />
        <br />
        <input
          id="gender"
          name="gender"
          placeholder="Enter your Gender"
          type="text"
          value=""
        />
        <br />
        <button
          id="button"
          type="submit"
          value="Submit"
        >
          ADD USER
        </button>
      </form>
    </div>
  </div>
  <div
    class="Table"
  >
    <div
      id="Table"
    >
      <input
        name="search"
        type="text"
        value=""
      />
      <table>
        <tbody>
          <tr>
            <th>
              NAME
            </th>
            <th>
              PHONE
            </th>
            <th>
              GENDER
            </th>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
